### ISSUE TITLE
Infinity Loss on Default Training

### ISSUE THREAD

**Issue by schulter** [tkipf/keras-gcn#11]
Hi,
I tried to run the package and after reading the other issues, I got it to work, using the other fork.
However, using the default parameters and the cora data set, I get an infinity loss:

`/home/sasse/.local/lib/python2.7/site-packages/kegra-0.0.1-py2.7.egg/kegra/utils.py:77: RuntimeWarning: divide by zero encountered in log
Epoch: 0001 train_loss= inf train_acc= 0.1643 val_loss= inf val_acc= 0.1600 time= 1.5656
Epoch: 0002 train_loss= inf train_acc= 0.1071 val_loss= inf val_acc= 0.1167 time= 0.0403
Epoch: 0003 train_loss= inf train_acc= 0.1643 val_loss= inf val_acc= 0.1600 time= 0.0467
Epoch: 0004 train_loss= inf train_acc= 0.1643 val_loss= inf val_acc= 0.1600 time= 0.0410
Epoch: 0005 train_loss= inf train_acc= 0.1643 val_loss= inf val_acc= 0.1600 time= 0.0405
Epoch: 0006 train_loss= inf train_acc= 0.1643 val_loss= inf val_acc= 0.1600 time= 0.0426
Epoch: 0007 train_loss= inf train_acc= 0.1643 val_loss= inf val_acc= 0.1600 time= 0.0462
Epoch: 0008 train_loss= inf train_acc= 0.1643 val_loss= inf val_acc= 0.1600 time= 0.0430
Epoch: 0009 train_loss= inf train_acc= 0.1643 val_loss= inf val_acc= 0.1600 time= 0.0420
/home/sasse/.local/lib/python2.7/site-packages/kegra-0.0.1-py2.7.egg/kegra/utils.py:77: RuntimeWarning: invalid value encountered in log
Epoch: 0010 train_loss= nan train_acc= 0.1571 val_loss= nan val_acc= 0.1167 time= 0.0531
Epoch: 0011 train_loss= inf train_acc= 0.1643 val_loss= inf val_acc= 0.1600 time= 0.0459
Epoch 11: early stopping
Test set results: loss= inf accuracy= 0.1400
`
This is probably because of the encountered zeros in `log` but I don't know why they occur.

**Comment by tkipf** [comment]
Thanks for catching this! I think keras 1.x is incompatible with newer versions of TensorFlow. To solve your issue, you can either:

1) Downgrade TensorFlow (0.12 should work),
2) Use Theano backend,
or 3) Use the solution proposed in this pull request: https://github.com/tkipf/keras-gcn/pull/10 - this Upgrades keras-gcn to keras-2.0.5 and thus works with the latest TensorFlow release. It breaks compatibility with Theano, however.

**Comment by schulter** [comment]
Hm, I did exactly step 3 already. I also tried to clone the forked repo by @buaahsh but that showed the same behavior. The code executes, but the results are incorrect.
The non-keras repo works fine, though.
I'm running `TF 1.0.1`, `Keras 2.0.6`

**Comment by tkipf** [comment]
You can still try the Theano backend or an earlier TensorFlow version. The argument order of some loss functions changed in TensorFlow at some point, which breaks optimization (such as in this case).

**Comment by schulter** [comment]
Okay, both solutions fix that. Thanks for the quick help.

**Comment by tkipf** [comment]
Glad it works! I'll try to fix Theano compatibility with the keras-2.0.5 pull request at some point. This should get rid of most of these issues.

**Comment by schulter** [comment]
That would be great. I'd really like to look into this.
The paper is great and very well explained by the way :-)

**Comment by tkipf** [comment]
Fixed with https://github.com/tkipf/keras-gcn/pull/10
